app/views/dashboard/saml_login.html.haml

```diff
@@ -0,0 +1,15 @@
+<!--[if IE 9]><html class="ie9 login-pf"><![endif]-->
+%html.login-pf{:class => get_vmdb_config[:server][:custom_login_logo] ? '' : 'rcue-login', :lang => I18n.locale.to_s.sub('-', '_')}
+  %head
+    = favicon_link_tag
+    = stylesheet_link_tag 'application'
+    = javascript_include_tag 'application'
+
+  %body
+    - if api_auth_token && validation_url
+      :javascript
+        sessionStorage.miq_token = '#{j_str api_auth_token}';
+        window.location.href = '#{j_str validation_url}';
+    - else
+      :javascript
+        window.location.href = '/dashboard/logout';
```

Comment:
I know we originally talked about doing `window.location.replace`, but did this not work out? I think we can also use just `window.location = '/relative_path_here'` (no `.href`) to get redirect-like behavior.

I think I'd try it without the `.href` and see if it behaves the way you want it to, if only to get rid of the extra characters 😉